tree/20_cfe_basics/ncf_lib.cf

```diff
@@ -772,3 +772,289 @@ body edit_defaults ncf_empty_select(select)
+      "args"               slist => { "${service}", "${action}" };
+
+  defaults:
+    # Rename "is-active" to "status" for service and init.d methods
+    # The condition is ((use service) or (use init.d))
+    # This allows staying compatible with any action and not be restricted to a predefined list
+    !systemctl_utility_present.!svcadm_utility_present.(service_utility_present|(!startsrc_utility_present.init_d_directory_present)))::
+      "action" string => "status", if_match_regex => "is-active";
+
+    # refresh is an alias to reload
+    any::
+      "action" string => "reload", if_match_regex => "refresh";
+
+  classes:
+
+    windows::
+      "is_valid_action" or => {
+                                strcmp("start", "${action}"),
+                                strcmp("stop", "${action}"),
+                                strcmp("restart", "${action}"),
+                              };
+      "is_restart_action" expression => strcmp("restart", "${action}");
+
+    any::
+      "is_boot_action"  or => { 
+                                strcmp("enable", "${action}"),
+                                strcmp("disable", "${action}"),
+                                strcmp("is-enabled", "${action}")
+                              };
+
+      "is_check_action" or => {
+                                strcmp("is-active", "${action}"),
+                                strcmp("is-active-process", "${action}"),
+                                strcmp("is-enabled", "${action}")
+                              };
+
+      "is_process_action" expression => strcmp("is-active-process", "${action}");
+
+    pass1::
+      "method_found"      expression => isvariable("method");
+
+    #####
+    # Actual command - for checks on non-windows systems
+    #####
+    pass1.is_check_action.!windows.!is_process_action.method_found::
+      "action_ok" expression => returnszero("${action_command} > /dev/null 2>&1", "useshell");
+
+    debian_8::
+      # This is to workaround a bug in Debian Jessie, that makes systemctl is-enabled
+      # fail if run against a service that still uses SysV style scripts
+      # See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=760616
+      "is_broken_action"    expression => strcmp("is-enabled", "${action}");
+      "init_script_present" expression => fileexists("/etc/init.d/${service}");
+      "broken_systemctl"    expression => "is_broken_action.init_script_present";
+
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  methods:
+
+    #####
+    # Actual command - for actions on non-windows systems, non using systemd
+    #####
+    pass2.is_check_action.!windows.!is_process_action.method_found.action_ok::
+      "force_success_class"    usebundle => _classes_success("${old_class_prefix}");
+      "force_success_class"    usebundle => _classes_success("${class_prefix}");
```

Comment:
we should have _classes_success_two (or equivalent) (this is not the place of this PR thought)